### ISSUE TITLE
(Windows 10) Fullscreen changes my display resolution when showing a dialog.

### ISSUE THREAD

**Issue by hugorosario** [fyne-io/fyne#1832]
### Describe the bug:

On windows 10, setting the app to fullscreen and then showing a dialog with a button click, will change my display resolution to whatever is defined in the window size. 
Resolution will only be restored after killing the app or switching to another app.

### To Reproduce:
Steps to reproduce the behaviour:

Set the window to fullscreen.
Create and show a dialog inside a button click function.

### Example code:
```go
func main() {
	myApp := app.New()
	myWindow := myApp.NewWindow("FullScreen Test")
	myWindow.Resize(fyne.NewSize(800, 600))
	myWindow.SetFullScreen(true)
	myWindow.SetContent(container.NewCenter(
		widget.NewButton("Show Dialog", func() {
			dialog.NewInformation("Fyne Information", "This is a fyne message dialog you got here.", myWindow).Show()
		})))
	myWindow.ShowAndRun()
}
```

### Device (please complete the following information):
 - **OS:** Windows 10 
 - **Version:** 20H2 build 19042.746
 - **Go version:** 1.15.6
 - **Fyne version:** 1.4.3

**Comment by andydotxyz** [comment]
Turns out this is not platform specific.
The code above replicates it despite the fyne_demo doing the same thing and not showing this issue.

**Comment by hugorosario** [comment]
I tested this with the fyne_demo in fullscreen mode and also have this issue and not only with dialogs, I just need to change to another section of the demo.

**Comment by andydotxyz** [comment]
I think I have found a patch. It's not going to make v2.0.0, but if you could test this locally we could get it in 2.0.1 and possibly back-port to 1.4.x.
[winfull.diff.txt](https://github.com/fyne-io/fyne/files/5836077/winfull.diff.txt)

**Comment by andydotxyz** [comment]
(FYI this seems to only happen when the dialog is larger than the window content, so it may go away if your application has more content)

**Comment by hugorosario** [comment]
> I think I have found a patch. It's not going to make v2.0.0, but if you could test this locally we could get it in 2.0.1 and possibly back-port to 1.4.x.
> [winfull.diff.txt](https://github.com/fyne-io/fyne/files/5836077/winfull.diff.txt)

Applied that patch to my package sources and it worked great.
Backporting this into the 1.4.x branch would be great.

Thank you.

**Comment by stanislav-janu** [comment]
I have same problem on MacOS and Raspbian... If I click button, then screen resolution has been changed.

fyne 2.0.0

code:
```
package main

import (
	"fmt"
	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/container"
	"fyne.io/fyne/v2/widget"
)

func main() {
	app := app.New()
	window := app.NewWindow("Inventura alkoholu")
	window.Resize(fyne.Size{Width: 800, Height: 480})
	window.SetFullScreen(true)

	var i = 0
	label := widget.NewLabel("Test")
	button := widget.NewButton("Test", func() {
		i++
		label.SetText(fmt.Sprintf("kliknuto %dx", i))
	})

	splitView := container.NewVSplit(label, button)
	window.SetContent(splitView)

	window.ShowAndRun()
}
```

**Comment by gmaisto** [comment]
Same here with fyne 2.0.3 on Windows 10. Dialogs from a fullscreen window change the screen resolution.

**Comment by andydotxyz** [comment]
Wow, this slipped through the cracks somehow. The fix will be in v2.0.4 and v2.1.0

**Comment by andydotxyz** [comment]
Available for testing on `develop` and `release/v2.0.x` branches.
